server/src/main/java/org/opensearch/indices/replication/common/CopyState.java

```diff
@@ -65,10 +62,6 @@ public CopyState(ReplicationCheckpoint requestedReplicationCheckpoint, IndexShar
     protected void closeInternal() {
         try {
             segmentInfosRef.close();
-            // commitRef may be null if there were no pending delete files
```

Comment:
Why are we removing the below code?

Reply:
The commit ref here is redundant because its now held inside of IE when getting the snapshot.

I think another refactor opportunity here is to also use this CopyState for remote store, and build it once after IE refreshes and store a ref to it.  This would eliminate the need for fetching & computing SI snapshots outside of refresh listeners.